### ISSUE TITLE
Restore runs out of memory

### ISSUE THREAD

**Issue by janmeier** [GreenmaskIO/greenmask#21]
I'm trying to restore a dump from our production database, but the restore command ends up being killed because it runs out of memory.

It happens at the same table each time - The machine has 8GB of memory, and every though the table is only 2GB according to `metadata.json`. The table has some large text columns (10k chars), so I'm not sure if that plays into it

My guess is that greenmask is loading the entire dump for the table into memory while restoring, but my go-fu is not strong enough to figure out if that's what's actually happening :/

**Comment by wwoytenko** [comment]
It might be memory leakage. Greenmask does not upload the whole table dump into memory, it writes this data by batch. So I suspect there might be only one problem - uncontrolled buffer growth.

Could you provide please:

* Greenmask version
* PostgreSQL version
* Table definition example (I will try to emulate your case)
* Do large columns contain a kind of json/html/xml?

**Comment by wwoytenko** [comment]
Were there important log lines before greenmask process was killed?

**Comment by janmeier** [comment]
> Were there important log lines before greenmask process was killed?

Nope, just "restoring table", and then after a minute or so `killed`

**Comment by janmeier** [comment]
> It might be memory leakage. Greenmask does not upload the whole table dump into memory, it writes this data by batch. So I suspect there might be only one problem - uncontrolled buffer growth.
> 
> Could you provide please:
> 
>    Greenmask version
0.1.5

>     PostgreSQL version
15.5

>     Table definition example (I will try to emulate your case)
```
postgres@127:app> \d "mealPlanTemplatesI18nOwned"
╒═════════════════╤══════════════════════════╤═════════════════════════════════════════╕
│ Column          │ Type                     │ Modifiers                               │
╞═════════════════╪══════════════════════════╪═════════════════════════════════════════╡
│ id              │ uuid                     │  not null                               │
├─────────────────┼──────────────────────────┼─────────────────────────────────────────┤
│ locale          │ locale                   │  not null                               │
├─────────────────┼──────────────────────────┼─────────────────────────────────────────┤
│ content         │ text                     │  not null                               │
├─────────────────┼──────────────────────────┼─────────────────────────────────────────┤
│ frontPageFileId │ uuid                     │                                         │
├─────────────────┼──────────────────────────┼─────────────────────────────────────────┤
│ createdAt       │ timestamp with time zone │  not null default statement_timestamp() │
├─────────────────┼──────────────────────────┼─────────────────────────────────────────┤
│ updatedAt       │ timestamp with time zone │                                         │
├─────────────────┼──────────────────────────┼─────────────────────────────────────────┤
│ name            │ character varying(255)   │                                         │
╘═════════════════╧══════════════════════════╧═════════════════════════════════════════╛
Indexes:
    "mealPlanTemplatesI18nOwned_pkey" PRIMARY KEY, btree (id, locale)
Foreign-key constraints:
    "mealplantemplatesi18nowned_id_foreign" FOREIGN KEY (id) REFERENCES "mealPlanTemplatesOwned"(id) ON UPDATE CASCADE ON DELETE CASCADE
Inherits: "mealPlanTemplatesI18n"

```
 
>    Do large columns contain a kind of json/html/xml?

Yes, they contain HTML

**Comment by janmeier** [comment]
I ran another restore, and the issue happened again, altough on a different table this time. The table doesn't have any big text columns like the previous one, but it is XX GB in size.

I was able to `pg_restore` the data without issue, so there seem to be a bug somewhere in greenmask.

This does mean that for now I can let greenmask handle the dump + anonymization, and then restore with `pg_restore` :man_dancing:

**Comment by janmeier** [comment]
> > Were there important log lines before greenmask process was killed?
> 
> Nope, just "restoring table", and then after a minute or so `killed`

```
2024-02-23T06:29:40Z DBG ../home/runner/work/greenmask/greenmask/internal/db/postgres/restore.go:377 > restoring objectName="table \"public\".\"meal_plan_meal\"" pid=37084 workerId=1
2024-02-23T06:29:40Z DBG ../home/runner/work/greenmask/greenmask/internal/db/postgres/restorers/table.go:64 > performing pgcopy statement copyStmt="COPY \"public\".\"meal_plan_meal\" (\"id\", \"meal_id\", \"meal_plan_id\", \"meal_group_id\", \"is_locked\", \"sort_weight\", \"created_at\", \"ignore_restrictions\", \"profileId\", \"scale\", \"ignore_macro_splits\", \"ignore_calories\", \"updatedAt\") FROM stdin" pid=37084
Killed
```

**Comment by wwoytenko** [comment]
The problem is found. I will let you know once the fix is released.

**Comment by wwoytenko** [comment]
@janmeier Thank you one more time for reporting this bug. 
It is fixed in v0.1.6 now.

**Comment by janmeier** [comment]
:racehorse: so fast, thanks a bunch! I can confirm that with the latest version, memory usage stays at a constant ~200mb, just like it did with `pg_restore` :heart_eyes_cat:
